src/base/torrentfilter.h

```diff
@@ -81,13 +81,13 @@ class TorrentFilter
     bool setCategory(const QString &category);
     bool setTag(const QString &tag);
 
-    bool match(BitTorrent::TorrentHandle *const torrent) const;
+    bool match(const BitTorrent::TorrentHandle *const torrent) const;
 
 private:
-    bool matchState(BitTorrent::TorrentHandle *const torrent) const;
-    bool matchHash(BitTorrent::TorrentHandle *const torrent) const;
-    bool matchCategory(BitTorrent::TorrentHandle *const torrent) const;
-    bool matchTag(BitTorrent::TorrentHandle *const torrent) const;
+    bool matchState(const BitTorrent::TorrentHandle *const torrent) const;
+    bool matchHash(const BitTorrent::TorrentHandle *const torrent) const;
+    bool matchCategory(const BitTorrent::TorrentHandle *const torrent) const;
+    bool matchTag(const BitTorrent::TorrentHandle *const torrent) const;
```

Comment:
technically, the "top-level" `const` is redundant in function declarations. (i.e. the right-most `const`)

Reply:
>the right-most const

err.. those are class functions.

Reply:
ah, i meant right-most on the parameter declaration. i.e.

```
bool matchTag(const BitTorrent::TorrentHandle *const torrent) const;
                                               ^^^^^
```

that const refers to that the `torrent` variable, can't be mutated. Which is great in the function *definition*, but in the declaration, the name of the variable isn't really significant, and that `const` has no meaning here.

Reply:
yeah, I thought so, however I'm not so sure how other team members thinks about it, I'll wait for their reviews before changing it.

Reply:
it all becomes much clearer by sticking to [continental const placement](https://kuhllib.com/2012/01/17/continental-const-placement/) (aka east const)

Reply:
>bool matchTag(const BitTorrent::TorrentHandle *const torrent) const;

Yes, the right `const` is applied to pointer so it is redundant and meaningless in the method declaration (as well as for other simple types). We've discussed this case before.